torch/_subclasses/functional_tensor.py

```diff
@@ -537,6 +559,29 @@ def unwrap(x):
                         torch.Tensor, wrap, outs_unwrapped
                     )
                 else:
+                    arg_mutation_counters = []
+
+                    if self.export:
+                        # Check args before the operation
+                        for arg in args:
+                            if isinstance(
+                                arg, FunctionalTensor
+                            ) and torch._is_functional_tensor(arg.elem):
```

Comment:
nit: we have pretty strong invariants that FunctionalTensor's inner tensors are always C++ `FunctionalTensor`s. So you can just remove the `_is_functional_tensor(arg.elem)` check